libvisual/tools/lv-tool/display/display.cpp

```diff
@@ -34,19 +35,17 @@ class Display::Impl
 
     std::unique_ptr<DisplayDriver> driver;
 
-    Impl ()
-        : driver (nullptr)
+    Impl (std::unique_ptr<DisplayDriver> driver_)
+        : driver {std::move (driver_)}
     {}
 
     ~Impl ()
     {}
 };
 
 Display::Display (std::string const& driver_name)
-  : m_impl (new Impl)
+    : m_impl {new Impl {DisplayDriverFactory::instance().make (driver_name, *this)}}
```

Comment:
@kaixiong we are passing `*this` here while `this` is in the middle of construction and not ready for any use beyond maybe holding a reference or pointer to it. We are sort-of doing that already before but… does we even need that? If it's out of scope, can it be resolved later?

Reply:
@hartwork good catch! I forgot about this but it's definitely safe with the current code since every driver class only stores the reference during their construction. But let me think about it more.